File: MarketFlutter/lib/models/token.dart


class Token{
  int id;
  String refreshToken;
  DateTime expiryDateTime;
  String userId;
  String? accessToken;

  Token({required this.id, required this.refreshToken, required this.expiryDateTime, required this.userId,  this.accessToken});

  factory Token.fromJson(Map<String, dynamic> json) {
    Token res = Token(
      id: json['id'] as int,
      refreshToken: json['refreshToken'] as String,
      expiryDateTime: DateTime.parse(json['expiryDateTime']),
      userId: json['userId'] as String,
      accessToken: json['accessToken'] as String,
    );
    return res;
  }

  // Method to convert User instance to a JSON map
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'refreshToken': refreshToken,
      'expiryDateTime': expiryDateTime.toString(),
      'accessToken': accessToken,
      'userId': userId,
    };
  }
}

File: MarketPortal/Market/Data/Models/Token.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Market.Data.Models
{
    public class Token
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string RefreshToken { get; set; }

        [Required]
        public DateTime ExpiryDateTime { get; set; }

        [Required]
        [ForeignKey(nameof(User))]
        public Guid UserId { get; set; }

        public User User { get; set; }

        [NotMapped]
        public string AccessToken { get; set; }


    }
}
